### PR TITLE
feat(server): Add credentials API endpoints

### LINKED ISSUE #8003: API endpoints to retrieve integration credentials
- Part of #8002

We need the following endpoints:
* `GET /integrations/{provider}/credentials` - list all credentials for a provider, without secrets (metadata only)
* `GET /integrations/{provider}/credentials/{cred_id}` - retrieve a set of credentials (including secrets)

## OAuth email access

1. Google, need scopes: `email`, (and likely) `openid`, `profile`<br>Accessed at: [https://www.googleapis.com/oauth2/v2/userinfo](https://www.googleapis.com/oauth2/v2/userinfo)<br>Useful [Stackoverflow discussion](https://stackoverflow.com/questions/24442668/google-oauth-api-to-get-users-email-address)
2. GitHub, scopes: `user:email`<br>[API docs](https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user)
3. Notion; scopes are set in [Capabilities](https://developers.notion.com/reference/capabilities)<br>Email is returned with access [token response object](https://developers.notion.com/docs/authorization#step-4-notion-responds-with-an-access_token-and-additional-information) at: `owner.person.email` for non-bot users

### PR DESCRIPTION
**Pull request by kcze**
- Part of #8002
- Resolves #8003

### Changes 🏗️

- Add two endpoints to OAuth `integrations.py`:
  - `GET /integrations/{provider}/credentials` - list all credentials for a provider, without secrets (metadata only)
   - `GET /integrations/{provider}/credentials/{cred_id}` - retrieve a set of credentials (including secrets)

- Add `username` property to `Credentials` types
   - Add logic to populate `username` in OAuth handlers

- Expand `CredentialsMetaResponse` and remove `credentials_` prefix from properties

- Fix `autogpt_libs` dependency caching issue

- Remove accidentally duplicated OAuth handler files in `autogpt_server/integrations`

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/server/integrations.py [hunk 1]
```diff
@@ -80,6 +84,85 @@ async def callback(
...
+    credential = store.get_creds_by_id(user_id, cred_id)
+    if not credential:
+        raise HTTPException(status_code=404, detail="Credentials not found")
+    if credential.provider != provider:
+        raise HTTPException(
+            status_code=400, detail="Credentials do not match the specified provider"
+        )
+
+    if isinstance(credential, OAuth2Credentials):
+        return CredentialsDetailResponse(
```
Comment (kcze) [inline rnd/autogpt_server/autogpt_server/server/integrations.py]: Should I split this into two models, one for each type?

File: rnd/autogpt_server/autogpt_server/server/integrations.py [hunk 2]
```diff
@@ -80,6 +84,85 @@ async def callback(
...
+    "/{provider}/credentials", response_model=list[CredentialsMetaResponse]
+)
+async def list_credentials(
+    provider: Annotated[str, Path(title="The provider to list credentials for")],
+    user_id: Annotated[str, Depends(get_user_id)],
+    store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
+) -> list[CredentialsMetaResponse]:
+    credentials = store.get_creds_by_provider(user_id, provider)
+    return [
+        CredentialsMetaResponse(credentials_id=cred.id, credentials_type=cred.type)
```
Comment (kcze) [inline rnd/autogpt_server/autogpt_server/server/integrations.py]: Do we need to return anything else here?

Comment (Pwuts) [reply rnd/autogpt_server/autogpt_server/server/integrations.py]: - Authorized scopes (if any/applicable)
- Some kind of human-readable identifier, e.g. a description or the login/e-mail of the linked account

File: rnd/autogpt_server/autogpt_server/server/integrations.py (hunk 1 above)
Comment (Pwuts) [reply rnd/autogpt_server/autogpt_server/server/integrations.py]: Why not just output the internal objects directly?

Comment (kcze) [reply rnd/autogpt_server/autogpt_server/server/integrations.py]: will do, I see secrets are dumped properly

File: rnd/autogpt_server/autogpt_server/server/integrations.py (hunk 2 above)
Comment (kcze) [reply rnd/autogpt_server/autogpt_server/server/integrations.py]: Done - scopes and email

File: rnd/autogpt_server/autogpt_server/server/integrations.py [hunk 3]
```diff
@@ -84,83 +85,44 @@ async def callback(
...
     store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
 ) -> list[CredentialsMetaResponse]:
-    credentials = store.get_creds_by_provider(user_id, provider)
+    credentials = store.get_creds_by_provider(auth["sub"], provider)
     return [
-        CredentialsMetaResponse(credentials_id=cred.id, credentials_type=cred.type)
+        CredentialsResponse(
+            credentials_id=cred.id,
+            credentials_type=cred.type,
+            user_email=auth["email"],
```
Comment (Pwuts) [inline rnd/autogpt_server/autogpt_server/server/integrations.py]: I meant the e-mail linked to the credentials, not the application user e-mail. Getting such metadata for credentials is different for every service, but for e.g. Notion it's stored in the credentials' metadata.

File: rnd/autogpt_server/autogpt_server/server/integrations.py [hunk 4]
```diff
@@ -84,83 +85,44 @@ async def callback(
...
     user_id: Annotated[str, Depends(get_user_id)],
     store: Annotated[SupabaseIntegrationCredentialsStore, Depends(get_store)],
-) -> CredentialsDetailResponse:
+) -> Credentials:
     credential = store.get_creds_by_id(user_id, cred_id)
     if not credential:
         raise HTTPException(status_code=404, detail="Credentials not found")
     if credential.provider != provider:
         raise HTTPException(
             status_code=400, detail="Credentials do not match the specified provider"
```
Comment (Pwuts) [inline rnd/autogpt_server/autogpt_server/server/integrations.py]: Should be 404 imo: within the scope of the given provider, no credentials exist with this ID.

**Review by Pwuts — APPROVED** [review]
LGTM!

**Pull request merged**